diff --git a/muaccounts/views.py b/muaccounts/views.py
index eb707f1..fe57f0f 100644
--- a/muaccounts/views.py
+++ b/muaccounts/views.py
@@ -15,6 +15,8 @@ from django.template.loader import render_to_string
 from forms import MUAccountCreateForm, MUAccountForm, AddUserForm
 from models import MUAccount
 
+from livesearch.models import SearchApi
+
 try: import sso
 except ImportError: USE_SSO = False
 else: USE_SSO = getattr(settings, 'MUACCOUNTS_USE_SSO', True)
@@ -99,7 +101,17 @@ def account_detail(request, return_to=None, extra_context={}):
     else:
         uform = AddUserForm()
 
-    ctx = dict(object=account, form=form, add_user_form=uform)
+    search_apis = SearchApi.objects.all()
+    if 'api_list' in request.POST:
+        for api in search_apis:
+          account.searchapis.remove(api)
+        apis = request.POST.getlist('api_list')
+        for api_id in apis:
+          api_obj = SearchApi.objects.get(id=api_id)
+          account.searchapis.add(api_obj)
+        account.save()
+
+    ctx = dict(object=account, form=form, add_user_form=uform, search_apis=search_apis)
     ctx.update(extra_context)
 
     return direct_to_template(
